spec/helpers/perf/drivers/docker.lua

```diff
@@ -197,7 +218,15 @@ function _M:start_upstream(conf)
 
   self.log.info("worker is started")
 
-  return "http://" .. worker_vip .. ":" .. UPSTREAM_PORT
+  if port_count == 1 then
```

Comment:
I think `start_upstream` sometimes returning 1 string and sometimes returning an array of strings makes it more difficult to use. I suggest removing the `if port_count == 1` condition and making it always return a table (which sometimes will have a single element). This will require tweaking on tests that assume that it returns a string, but that seems like a simple change to make.

Reply:
that's a good idea.